### PR TITLE
add dicehub image(one provider in dicehub)

### PR DESCRIPTION
**Pull request by ZLValien**
#### What type of this PR

/kind feature

#### What this PR does / why we need it:

add provider of dicehub image

### CODE REVIEW

**Comment by sfwn** [comment]
Make all of your commits verified. Setup your committer&author correctly.

File: modules/dicehub/image/provider.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+	"github.com/jinzhu/gorm"
+
+	logs "github.com/erda-project/erda-infra/base/logs"
+	servicehub "github.com/erda-project/erda-infra/base/servicehub"
+	transport "github.com/erda-project/erda-infra/pkg/transport"
+	pb "github.com/erda-project/erda-proto-go/dicehub/image/pb"
+	db "github.com/erda-project/erda/modules/dicehub/image/db"
+)
+
+type config struct {
```
Comment (sfwn) [inline modules/dicehub/image/provider.go:26]: move env configs from pkg conf to here.

File: modules/dicehub/image/provider.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+}
+
+// +provider
+type provider struct {
+	Cfg          *config
+	Log          logs.Logger
+	Register     transport.Register `autowired:"service-register" optional:"true"`
```
Comment (sfwn) [inline modules/dicehub/image/provider.go:33]: service-register is required here.

**Review by sfwn — CHANGES_REQUESTED** [review]

File: modules/dicehub/image/provider.go (hunk 2 above)
Comment (ZLValien) [reply modules/dicehub/image/provider.go]: done

File: modules/dicehub/image/provider.go (hunk 1 above)
Comment (ZLValien) [reply modules/dicehub/image/provider.go:26]: image does not have its own config

**Comment by Effet** [comment]
I know `dicehub image` means `docker image` related apis in dicehub provided.

can you polish your pr title?

File: conf/dicehub/dicehub.yaml [hunk 3]
```diff
@@ -1,4 +1,22 @@
...
+grpc-server:
+  addr: ":7080"
+
+service-register:
+
+erda.dicehub.image:
+
+mysql:
+  host: "${MYSQL_HOST:localhost}"
+  port: ${MYSQL_PORT:3306}
```
Comment (sfwn) [inline conf/dicehub/dicehub.yaml:16]: quote it uniformly

File: modules/dicehub/dbclient/image.go [hunk 4]
```diff
@@ -54,15 +54,6 @@ func (client *DBClient) GetImagesByRelease(releaseID string) ([]Image, error) {
...
-func (client *DBClient) GetImageByID(imageID int64) (*Image, error) {
-	var image Image
-	if err := client.Where("id = ?", imageID).Find(&image).Error; err != nil {
-		return nil, err
-	}
-	return &image, nil
-}
-
 // GetImageByImage 根据 image 获取镜像
 func (client *DBClient) GetImageByImage(image string) (*Image, error) {
```
Comment (sfwn) [inline modules/dicehub/dbclient/image.go:58]: Migrate all image related functions together.

File: modules/dicehub/image/db/table.go [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+	ImageTag  string `json:"imageTag" gorm:"type:varchar(64)"`            // image tag
+	Image     string `json:"image" gorm:"not null"`                       // image addr
+}
+
+// Set table name
+func (Image) TableName() string {
+	return "ps_images"
+}
+
+
```
Comment (sfwn) [inline modules/dicehub/image/db/table.go:31]: lint failed.

**Review by sfwn — CHANGES_REQUESTED** [review]

File: modules/dicehub/dbclient/image.go (hunk 4 above)
Comment (ZLValien) [reply modules/dicehub/dbclient/image.go:58]: it is used by release provider. not image provider

**Pull request closed without merging**